presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorMetadata.java

```diff
@@ -186,17 +191,41 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
     {
         RaptorTableHandle handle = checkType(tableHandle, RaptorTableHandle.class, "tableHandle");
         SchemaTableName tableName = new SchemaTableName(handle.getSchemaName(), handle.getTableName());
-        List<ColumnMetadata> columns = dao.listTableColumns(handle.getTableId()).stream()
+        List<TableColumn> tableColumns = dao.listTableColumns(handle.getTableId());
+        if (tableColumns.isEmpty()) {
+            throw new PrestoException(RAPTOR_ERROR, "Table does not have any columns: " + tableName);
```

Comment:
Actually, this would happen if the table is dropped between calls, so just make this
```java
throw new TableNotFoundException(tableName);